openarc/util.c

```diff
@@ -891,9 +892,9 @@ arcf_mkarray(char *in)
 	if (out == NULL)
 		return (const char **) NULL;
 
-	for (p = strtok(in, ","); *p != '\0'; p = strtok(NULL, ","))
-		out[c] = p;
-	out[c] = NULL;
+	for (p = strtok_r(in, ",", &ctx); p != NULL; p = strtok_r(NULL, ",", &ctx))
+		out[c++] = p;
+	out[n] = NULL;
```

Comment:
So if there's a really strong consensus that this is confusing, then ok.  But "it's confusing" != "it's wrong"

That said, the reuse of loop variables as above is a classic source of errors.  The code would be safer if the loop variable were scoped exclusively to the loop in which it's used.

`n` is required for the initial malloc.  It can't be dropped altogether.

By using `n`, which is the variable used in the corresponding malloc, rather than a loop variable that is only related based on behavior, we help ensure against memory errors.

And I don't understand why it would be necessary or desirable to reset `c` in either case, since it's a stack variable used exclusively for this loop.